src/config.cc

```diff
@@ -73,16 +81,34 @@ void ParseCommadLineOptions(int argc, const char *argv[], /* out */Config& conf)
 
     parser.parse_check(argc, argv);
 
-    ValidateParsedCommandLine(parser);
 
     if (!parser.rest().empty()) {
         conf.input = parser.rest().front();
     }
 
-    conf.lineNumbers = parser.exist(config::UseLineNumbers);
-    conf.validate    = parser.exist(config::Validate);
-    conf.format      = parser.get<std::string>(config::Format);
-    conf.astType     = parser.get<std::string>(config::Type);
-    conf.output      = parser.get<std::string>(config::Output);
-    conf.sourceMap   = parser.get<std::string>(config::Sourcemap);
+    conf.lineNumbers      = parser.exist(config::UseLineNumbers);
+    conf.validate         = parser.exist(config::Validate);
+    conf.format           = parser.get<std::string>(config::Format) == "yaml" ? drafter::YamlFormat : drafter::JsonFormat;
+    conf.astType          = parser.get<std::string>(config::Type) == "ast" ? drafter::NormalASTType : drafter::RefractASTType ;
+    conf.output           = parser.get<std::string>(config::Output);
+    conf.sourceMap        = parser.get<std::string>(config::Sourcemap);
+    conf.refractSourceMap = parser.exist(config::RefractSourceMap);
+
+
+    //  source map generating logic 
+    //  (s - sourcemap, r - refract sourcemap)
+    //  ast type: | AST | REFRACT
+    //  ----------+-----+---------
+    //  opt: -s   | s,r | r
```

Comment:
I don't understand this line. According to the logic below, I thought Refract supports `s,r` and AST supports `s` only.